pytorch_lightning/loops/dataloader/evaluation_loop.py

```diff
@@ -181,11 +181,13 @@ def on_run_end(self) -> List[_OUT_DICT]:
         logged_outputs, self._logged_outputs = self._logged_outputs, []  # free memory
         # include any logged outputs on epoch_end
         epoch_end_logged_outputs = self.trainer._logger_connector.update_eval_epoch_metrics()
+        logger_metrics = dict(ChainMap(*logged_outputs))
+        logger_metrics.update(epoch_end_logged_outputs)
         for dl_outputs in logged_outputs:
             dl_outputs.update(epoch_end_logged_outputs)
 
         # log metrics
-        self.trainer._logger_connector.log_eval_end_metrics()
+        self.trainer._logger_connector.log_eval_end_metrics(logger_metrics)
```

Comment:
Wouldn't this work?

```suggestion
        all_logged_outputs = dict(ChainMap(*logged_outputs))  # list[dict] -> dict  
        self.trainer._logger_connector.log_eval_end_metrics(all_logged_outputs)
```

Reply:
it will, but since epoch end metrics are duplicated here, I think the current one might be a little better performance-wise.